### PR TITLE
DDF 3361 Restrict xStream's Deserializable Objects

### PR DESCRIPTION
**Pull request by jefchng**
#### What does this PR do?
`TransactionMessageBodyReader`, `FeatureCollectionMessageBodyReaderWfs10`, and `FeatureCollectionMessageBodyReaderWfs20` deserialized XML java objects without restricting which classes it should deserialize. xStream's `allowTypes()` method is now used to restrict deserializations to desired types only. 

#### Who is reviewing it? 
@garrettfreibott 
@tbatie 

#### Choose 2 committers to review/merge the PR. 
(please choose ONLY two committers from below, delete the rest)
@figliold 
@coyotesqrl 

#### How should this be tested? (List steps with links to updated documentation)
Manually verify that data can still be queried through the CSW endpoint and both of the WFS sources.
 
#### What are the relevant tickets?
[DDF-3361](https://codice.atlassian.net/browse/DDF-3361)

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests
#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by tbatie** [comment]
@stustison should take a look at this before merge.

File: catalog/spatial/csw/spatial-csw-endpoint/src/test/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/reader/TestTransactionMessageBodyReader.java [hunk 1]
```diff
@@ -282,6 +283,17 @@
           + "    </csw:Update>\n"
           + "</csw:Transaction>";
 
+  private static final String DYNAMIC_PROXY_SERIALIZED_XML =
```
[the remark this answers is not in the document]
Comment (garrettfreibott) [reply catalog/spatial/csw/spatial-csw-endpoint/src/test/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/reader/TestTransactionMessageBodyReader.java]: We were trying to match the style of the tests that were already in this class - there are like 20 other small test data strings above this line.

**Comment by bdeining** [comment]
@jefchng needs rebase

**Comment by tbatie** [comment]
build now

**Comment by tbatie** [comment]
build now

**Pull request merged**